### PR TITLE
Refactor Sessions

### PR DESCRIPTION
**Pull request by rhnonose**
- Adicionar testes ao session controller
- Refatorar workflow de autenticação para simplificar

### CODE REVIEW

File: lib/re/accounts/users.ex [hunk 1]
```diff
@@ -0,0 +1,15 @@
+defmodule Re.Accounts.Users do
```
Comment (gusaiani) [inline lib/re/accounts/users.ex]: Modules should have a @moduledoc tag.

**Comment by gusaiani** [comment]
Ebert has finished reviewing this Pull Request and has found:

* 2 possible new issues (including those that may have been commented here).
* 1 fixed issue! 🎉

But beware that this branch is **[3 commits behind the `em-casa:master` branch](https://github.com/em-casa/re/compare/em-casa:nonose/session_controller_tests...em-casa:master)**, and a review of an up to date branch would produce more accurate results.

You can see more details about this review at https://ebertapp.io/github/em-casa/re/pulls/64.

File: lib/re_web/router.ex [hunk 2]
```diff
@@ -34,7 +34,7 @@ defmodule ReWeb.Router do
       resources "/images", ImageController, only: [:index, :create, :delete]
     end
     put "/listings/:listing_id/image_order", ListingController, :order
-    resources "/users", UserController, except: [:new, :edit, :create]
+    # resources "/users", UserController, except: [:new, :edit, :create]
```
Comment (gusaiani) [inline lib/re_web/router.ex]: @rhnonose, deletamos essa linha comentada?

Comment (rhnonose) [reply lib/re_web/router.ex]: Opa esqueci.

**Comment by gusaiani** [comment]
Ebert has finished reviewing this Pull Request and has found:

* 1 possible new issue (including those that may have been commented here).
* 1 fixed issue! 🎉

But beware that this branch is **[3 commits behind the `em-casa:master` branch](https://github.com/em-casa/re/compare/em-casa:nonose/session_controller_tests...em-casa:master)**, and a review of an up to date branch would produce more accurate results.

You can see more details about this review at https://ebertapp.io/github/em-casa/re/pulls/64.

**Comment by gusaiani** [comment]
Ebert has finished reviewing this Pull Request and has found:

* 1 possible new issue (including those that may have been commented here).
* 1 fixed issue! 🎉

You can see more details about this review at https://ebertapp.io/github/em-casa/re/pulls/64.

**Comment by gusaiani** [comment]
Ebert has finished reviewing this Pull Request and has found:

* 1 fixed issue! 🎉

You can see more details about this review at https://ebertapp.io/github/em-casa/re/pulls/64.

**Pull request merged**